### PR TITLE
[TVMSCRIPT] Improve tvmscript type hints

### PR DESCRIPTION
**Pull request by tkonolige**
- Change numeric types to classes so they work as function arguments.
- Add var as a class.
- Add floordiv, index, and mod to PrimExpr. (I can't figure out a way to change the type of `range`, `__index__` is a stopgap to make type checking not complain).

@wrongtest @Hzfengsy @vinx13

### CODE REVIEW

File: python/tvm/script/tir/__init__.pyi [hunk 1]
```diff
@@ -299,7 +284,7 @@ special_stmt - Annotations
 """
 
 def buffer_var(dtype: str, storage_scope: str) -> Var: ...
-def func_attr(attrs: Mapping[str, Object]) -> None: ...
+def func_attr(attrs: Mapping[str, Object | str | bool | int | float]) -> None: ...
```
Comment (vinx13) [inline python/tvm/script/tir/__init__.pyi:287]: `|` syntax is only available in python 3.10, we may prefer using `Union` for now

**Review by vinx13 — COMMENTED** [review]
Thanks for the PR. Seems this will fix https://github.com/apache/tvm/issues/11192 cc @yelite

File: python/tvm/script/tir/__init__.pyi (hunk 1 above)
Comment (tkonolige) [reply python/tvm/script/tir/__init__.pyi:287]: Huh, it worked with my python 3.8. But I've switched this to a `Union`

**Comment by tkonolige** [comment]
I'm not sure this will fix #11192

**Comment by junrushao1994** [comment]
This is a good PR, and don't worry about whether or not it's going to fix #11192

**Review by vinx13 — APPROVED** [review]

**Pull request merged**